0847_deducing_this/deducing-this.bs

```diff
@@ -1360,6 +1342,77 @@ auto not_fn(F&& f) {
+
+## Function Pointer Types ## {#faq-function-ptr-type}
+
+A valid question to ask is what should be the type of this-annotated functions that have a member function equivalent? There are only two options, each with a trade-off. Please assume the existence of these three functions:
+
+```cpp
+struct Y {
+    int f(int, int) const&;         // exists
+    int g(this Y const&, int, int); // this paper
+    int h(this Y, int, int);        // this paper, by value
+};
+```
+
+`g` has a current equivalent (`f`), while `h` does not. `&Y::h`'s type *must* be a regular function pointer.
+
+If we allow `g`'s type to be a pointer-to-member-function, we get non-uniformity between the types of `h` and `g`. We also get implementation issues because the types a template can result in are non-uniform (is this a template for a member function or a free function? Surprise, it's both!).
+
+We also get forward compatibility with any concievable proposal for extension methods - those will *also* have to be free functions by necessity, for roughly the same reasons.
+
+The paper originally proposed it the other way, but this was changed to the current wording through EWG input in Cologne, 2018.
+
+## Deducing to Base-Class Pointer ## {#faq-computed-deduction}
+
+One of the pitfalls of having a deduced object parameter is when the intent is solely to deduce the *cv*-qualifiers and value category of the object parameter, but a derived type is deduced as well &mdash; any access through an object that might have a derived type could inadvertently refer to a shadowed member in the derived class. While this is desirable and very powerful in the case of mixins, it is not always desirable in other situations. Superfluous template instantiations are also unwelcome side effects.
+
+One family of possible solutions could be summarized as **make it easy to get the base class pointer**. However, all of these solutions still require extra instantiations. For `optional::value()`, we really only want four instantiations: `&`, `const&`, `&&`, and `const&&`. If something inherits from `optional`, we don't want additional instantiations of those functions for the derived types, which won't do anything new, anyway. This is code bloat.
+
+*This is already a problem for free-function templates*: The authors have heard many a complaint about it from library vendors, even before this paper was introduced, as it is desirable to only deduce the ref-qualifier in many contexts. Therefore, it might make sense to tackle this issue in a more general way. A complementary feature could be proposed to constrain *type deduction*.
+
+The authors strongly believe this feature is orthogonal. However, hoping that mentioning that solutions are in the pipeline helps gain consensus for this paper, we mention one solution here. The proposal is in early stages, and is not in the mailing. It will be presented in the next mailing: [computed deduction](https://atomgalaxy.github.io/isocpp-1107/D1107.html)
```

Comment:
Which next mailing? Post-Belfast?

Reply:
Probably should write "is not in the pre-Belfast mailing" and clarify which one the next one is (for readers of this paper a month+ from now.

Reply:
Good point. Fixed. Merge? Then we can attach to the EWG queue.